aiida_common_workflows/cli/plot.py

```diff
@@ -1,34 +1,42 @@
 # -*- coding: utf-8 -*-
 """Commands to plot results from a workflow."""
-
 import click
 
 from aiida.cmdline.params import arguments
 from aiida.cmdline.utils import echo
+from aiida.plugins import WorkflowFactory
 
 from .root import cmd_root
 from . import options
 
+EquationOfStateWorkChain = WorkflowFactory('common_workflows.eos')
+DissociationCurveWorkChain = WorkflowFactory('common_workflows.dissociation_curve')
+
 
 @cmd_root.group('plot')
 def cmd_plot():
     """Plot results from a workflow."""
 
 
 @cmd_plot.command('eos')
-@arguments.NODE()
+@arguments.WORKFLOW()
 @options.PRECISIONS()
 @options.PRINT_TABLE()
 @options.OUTPUT_FILE()
-def cmd_plot_eos(node, precisions, print_table, output_file):
+def cmd_plot_eos(workflow, precisions, print_table, output_file):
     """Plot the results from an `EquationOfStateWorkChain`."""
     # pylint: disable=too-many-locals
     from tabulate import tabulate
 
     from aiida.common import LinkType
     from aiida_common_workflows.common.visualization.eos import get_eos_plot
 
-    outputs = node.get_outgoing(link_type=LinkType.RETURN).nested()
+    if workflow.process_class is not EquationOfStateWorkChain:
+        echo.echo_critical(f'Provided workflow with PK {workflow.pk} does not represent an EquationOfStateWorkChain.')
+    if workflow.exit_status != 0:
+        echo.echo_critical(f'Provided workflow with PK {workflow.pk} did not finish successfully.')
```

Comment:
In most `verdi` commands we use the following syntax to represent nodes
```suggestion
        echo.echo_critical(f'node {workflow.__class__.__name__}<{workflow.pk}> does not correspond to an EquationOfStateWorkChain.')
    if workflow.exit_status != 0:
        echo.echo_critical(f'workflow {workflow.__class__.__name__}<{workflow.pk}> did not finish successfully.')
```